tasks.c

```diff
@@ -58,16 +58,42 @@
     #include <stdio.h>
 #endif /* configUSE_STATS_FORMATTING_FUNCTIONS == 1 ) */
 
-#if ( configNUMBER_OF_CORES == 1 )
-    #if ( configUSE_PREEMPTION == 0 )
+#if ( configUSE_PREEMPTION == 0 )
 
 /* If the cooperative scheduler is being used then a yield should not be
  * performed just because a higher priority task has been woken. */
-        #define taskYIELD_IF_USING_PREEMPTION()
-    #else
-        #define taskYIELD_IF_USING_PREEMPTION()    portYIELD_WITHIN_API()
-    #endif
-#endif /* if ( configNUMBER_OF_CORES == 1 ) */
+    #define taskYIELD_IF_USING_PREEMPTION( pxTCB )
+    #define taskYIELD_FOR_TASK_IF_USING_PREEMPTION( pxTCB )
+#else
+
+    #if ( configNUMBER_OF_CORES == 1 )
+
+/* pxTCB is not used since the task can only runs on core 0 in single core. */
+        #define taskYIELD_IF_USING_PREEMPTION( pxTCB )    portYIELD_WITHIN_API()
```

Comment:
Why is the parameter needed here?  If it is needed it may need to be defined as follows to prevent warnings about the parameter not being used: { portYIELD_WITHIN_API(); ( void ) pxTCB; }

Reply:
This suggestion makes sense.
If taskYIELD_IF_USING_PREEMPTION is used in the following code, there will be compiler warning.
```c
void func( TCB_t * pxTCB )
{
    /* The function assumes pxTCB is used in this macro. However, this macro doesn't use pxTCB in
     * single core scheduler. There will be compiler warning. It is better taskYIELD_IF_USING_PREEMPTION
     * is defined as suggestion. */
    taskYIELD_IF_USING_PREEMPTION( pxTCB );
}
```
The macro is updated and more comments to explain why the pxTCB is not used in this macro.